### PR TITLE
RM-23161 Add Simple Graph Implementation/tests

### PR DESCRIPTION
**Pull request by JulianGriggs**
Hello -- This repository has been very helpful to me as I'm beginning to learn Go. With the dual purpose of getting practice and contributing back in some small way, I have added a Simple Graph implementation. A Simple Graph is an undirected graph without parallel edges or self loops. Let me know what you think.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

File: graph/simple.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	adj := make([]interface{}, deg)
+	i := 0
+	for key := range g.adjacencyList[v] {
+		adj[i] = key
+		i++
+	}
+	return adj, nil
+}
+
+func (g *SimpleGraph) Degree(v interface{}) (int, error) {
```
Comment (dustinhiatt-wf) [inline graph/simple.go]: The public methods should have docstrings.

**Comment by dustinhiatt-wf** [comment]
Once the docstrings are added, lgtm

**Comment by dustinhiatt-wf** [comment]
@JulianGriggs You mind adding some docstrings?

**Comment by JulianGriggs** [comment]
Sorry I forgot about this PR. I just added the requested docstrings.

**Comment by dustinhiatt-wf** [comment]
+1

**Comment by brianshannan-wf** [comment]
+1

**Pull request merged**